File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.cpp
class Solution {
public:
    int find(int a, int b){
        if(b==0)
            return a;
        
        return find(b,a%b);
    }

    int findGCD(vector<int>& nums) {
        int maxi=nums[0];
        int mini=nums[0];

        for(int i : nums){
            maxi=max(maxi,i);
            mini=min(mini,i);
        }

        return find(maxi,mini);

    }
};

File: 1979-find-greatest-common-divisor-of-array/1979-find-greatest-common-divisor-of-array.java
class Solution {
    public static int gcd(int a, int b){
        if(a==0) return b;
        return gcd(b%a, a);
    }
    public int findGCD(int[] nums) {
        int mini = nums[0];
        int maxi = nums[0];

        for(int i=0;i<nums.length;i++){
            mini = Math.min(nums[i],mini);
            maxi = Math.max(nums[i],maxi);
        }

        return gcd(maxi,mini);
    }
}